scipy/stats/_fit.py

```diff
@@ -782,12 +782,16 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
         to the Monte Carlo samples drawn from the null-hypothesized
         distribution. The purpose of these `guessed_params` is to be used as
         initial values for the numerical fitting procedure.
-    statistic : {"ad", "ks", "cvm", "filliben"}, optional
+    statistic : {"ad", "ks", "cvm", "filliben"} or callable, optional
         The statistic used to compare data to a distribution after fitting
         unknown parameters of the distribution family to the data. The
         Anderson-Darling ("ad") [1]_, Kolmogorov-Smirnov ("ks") [1]_,
         Cramer-von Mises ("cvm") [1]_, and Filliben ("filliben") [7]_
-        statistics are available.
+        statistics are available.  Alternatively, a callable with signature
+        ``(dist, data) ``may be supplied to compute the statistics.  Here
+        ``dist`` is a frozen dist object of one or more fitted distributions,
+        and ``data`` is an array containing one or more samples.  The callable
```

Comment:
To keep the documentation consistent, I reserve the plural of "sample" for multi-sample statistics like `ttest_ind`. When we pass in an N-dimensional array to a one-sample statistic like `ttest_1samp`, I'd say we're making a vectorized call to a one sample statistic.

It is not wrong to say "samples" here in regular English. But considering the ambiguity in the word "samples" - which people would even use interchangeably with a single sample with multiple "observations", I think it's important to limit our use of the language for clarity.

If we include the language above about computing the statistic along the provided `axis` of `data`, I think we can avoid saying "samples" but still be clear.